File: case01/Livro.py
class Livro:
    def __init__(self, titulo: str, autor: str, quantPag: int, preco: float, ano =None):
        if ano is not None:
            self.titulo = titulo
            self.autor = autor
            self.ano = ano
            self.quantPag = quantPag
            self.preco = preco
        else:
            self.titulo = titulo
            self.autor = autor
            self.quantPag = quantPag
            self.preco = preco

    def get_titulo(self):
        return self.titulo
    
    def get_autor(self):
        return self.autor
    
    def get_ano(self):
        return self.ano
    
    def get_quantPag(self):
        return self.quantPag
    
    def get_preco(self):
        return self.preco
    
    def set_titulo(self, titulo):
        self.titulo = titulo

    def set_autor(self, autor):
        self.autor = autor

    def set_quantPag(self, quantPag):
        self.quantPag = quantPag
    
    def set_preco(self, preco):
        self.preco = preco
        
    def calcularPrecoPorPagina(self):
        return self.quantPag/self.preco
    
    def __str__(self):
        return f"Título: {self.titulo}, Autor: {self.autor}, Ano: {self.autor}"

File: case01/java/Livro.java
public class Livro
{
    private String titulo, autor;
    private int ano, quantPag;
    private double preco;
    
    public Livro(String titulo, String autor, int ano, int quantPag, double preco){
        this.titulo = titulo;
        this.autor = autor;
        this.ano = ano;
        this.quantPag = quantPag;
        this.preco = preco;
    }
    
    public Livro(String titulo, String autor, int quantPag, double preco){
        this.titulo = titulo;
        this.autor = autor;
        this.quantPag = quantPag;
        this.preco = preco;
    }
    
    public String getTitulo(){
        return titulo;
    }
    public String getAutor(){
        return autor;
    }
    public int getAno(){
        return ano;
    }
    public int getQuantPag(){
        return quantPag;
    }
    public double getpreco(){
        return preco;
    }
    public void setTitulo(String titulo){
        this.titulo = titulo;
    }
    public void setAutor(String autor){
        this.autor = autor;
    }
    public void setAno(int ano){
        this.ano = ano;
    }
    public void setQuantPag(int quantPag){
        this.quantPag = quantPag;
    }
    public void sePpreco(double preco){
        this.preco = preco;
    }
    
    public double calcularPrecoPorPagina(){
        return quantPag/preco;
    }
     public String toString(){
         return "Título: " +titulo+
         ", Autor: " +autor+
         ", Ano: " +ano+
         ", \nQuantidade de Páginas: " +quantPag+
         ", Preço: " +preco+
         ", Preço por página: " +calcularPrecoPorPagina();
     }
}
